nmdc_server/storage.py

```diff
@@ -0,0 +1,170 @@
+from datetime import datetime, timedelta, timezone
+from enum import StrEnum
+from functools import cached_property
+from typing import Any, Iterable
+
+from google.auth.credentials import AnonymousCredentials
+from google.cloud.exceptions import NotFound
+from google.cloud.storage import Blob, Bucket, Client
+
+from nmdc_server.config import settings
+from nmdc_server.schemas import SignedUrl
+
+
+class BucketName(StrEnum):
+    """Enum for GCS bucket names"""
+
+    SUBMISSION_IMAGES = "nmdc-submission-images"
```

Comment:
Does each deployment of `nmdc-server` need more than 1 bucket? Should the bucket name be configurable via env var instead of hard-coded here?

Reply:
All deployments share one bucket for submission images.

Setting up a new bucket on the real GCS server requires elevated permissions to create it and set the CORS configuration. That is why the `nmdc-server` code _doesn't_ do it automatically. For now, setting up a bucket is a manual operation; it will be done very rarely. Later, it could be automated via helm charts. Either way, it's outside the scope of what `nmdc-server` should worry about.

Since all deployments share one bucket, objects coming into the bucket from different environments are differentiated by a prefix on the object name. That is configured via an environment variable (namely `NMDC_GCS_OBJECT_NAME_PREFIX`).

It's possible there will be other buckets for other purposes (other than submission images, that is) in the future. The enum is maybe a bit overkill for now, but it allows for future expansion.

Reply:
Thanks for clarifying. I think its probably a _good_ thing that the server code doesn't do any bucket creation itself.

My concern is more with keeping deployment configuration separate from application code. Even if all deployments share a bucket now, and its not likely to change in the future, I think its cleaner to make this (at least the string value) configurable through an environment variable. Totally fine to keep the enum, but maybe change to:

```suggestion
class BucketName(StrEnum):
    """Enum for GCS bucket names"""

    SUBMISSION_IMAGES = settings.gcs_submission_images_bucket
```